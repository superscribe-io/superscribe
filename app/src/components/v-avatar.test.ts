import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import VAvatar from './v-avatar.vue';

test('Mount component', () => {
	expect(VAvatar).toBeTruthy();

	const wrapper = mount(VAvatar, {
		slots: {
			default: 'Slot Content',
		},
	});

	expect(wrapper.html()).toMatchSnapshot();
});

test('tile prop', () => {
	const wrapper = mount(VAvatar, {
		props: {
			tile: true,
		},
	});

	expect(wrapper.classes()).toContain('tile');
});

test('small prop', () => {
	const wrapper = mount(VAvatar, {
		props: {
			small: true,
		},
	});

	expect(wrapper.classes()).toContain('small');
});
