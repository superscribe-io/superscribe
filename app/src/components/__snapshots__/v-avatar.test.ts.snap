// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `"<div data-v-8dd7c482=\\"\\" class=\\"v-avatar\\">Slot Content</div>"`;
